src/scripts.c

```diff
@@ -51,7 +51,11 @@ static void update_triggers(void)
 {
 	__attribute__((unused)) int ret = 0;
 
-	ret = system("/usr/bin/systemctl --no-block daemon-reload");
+	if (need_systemd_reexec) {
+		ret = system("/usr/bin/systemctl daemon-reexec");
+	}
```

Comment:
agreed. The reload is also a fairly fast and deterministic thing (it can only take so long to re-parse the changed units).

Adding --no-block here could mess things up, especially for `reload`